src/internal/Subscriber.ts

```diff
@@ -160,6 +160,7 @@ export class SafeSubscriber<T> extends Subscriber<T> {
               error?: ((e?: any) => void) | null,
               complete?: (() => void) | null) {
     super();
+    this.add(this._teardown);
```

Comment:
suggestion: Use `this._teardown.bind(this)` here rather than a per-subscriber `_teardown` property. That is, make `_teardown` a method rather than an instance property. Are there perf benefits with using the arrow function? Won't allocating the function per-subscriber be more expensive than the `bind`?

Reply:
Using `bind` also [allocates a new function per subscriber](https://github.com/v8/v8/blob/4b9b23521e6fd42373ebbcb20ebe03bf445494f9/src/builtins/builtins-function.cc#L199-L201).

In most previous iterations of V8, it was also slower, overall. I think that's still the case, just by a narrower margin. In general, IMO, we should avoid using `bind` where possible. Arrow functions, simple closures, call, and apply are better alternatives.